lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -0,0 +1,124 @@
+import fnmatch
+import ufe
+import mayaUsd.ufe as mu
+import maya.internal.common.ufe_ae.template as ufeAeTemplate
+
+# We manually import all the classes which have a 'GetSchemaAttributeNames'
+# method so we have access to it and the 'pythonClass' method.
+from pxr import Usd, UsdGeom, UsdLux, UsdRender, UsdRi, UsdShade, UsdSkel, UsdUI, UsdVol
+
+# SchemaBase template class for categorization of the attributes.
+class AETemplate(ufeAeTemplate.Template):
+    '''
+    This system of schema inherits groups attributes per schema helping the user
+    learn how attributes are stored.
+    '''
+    def __init__(self, ufeSceneItem):
+        # Get the UFE Attributes interface for this scene item.
+        self.attrS = ufe.Attributes.attributes(ufeSceneItem)
+
+        # And finally call our base class init which will in turn
+        # build the UI.
+        super(AETemplate, self).__init__(ufeSceneItem)
+
+    def createSection(self, layoutName, attrList, collapse=False):
+        # We create the section named "layoutName" if at least one
+        # of the attributes from the input list exists.
+        for attr in attrList:
+            if self.attrS.hasAttribute(attr):
+                with ufeAeTemplate.Layout(self, layoutName, collapse):
+                    # Note: addControls will silently skip any attributes
+                    #       which do not exist.
+                    self.addControls(attrList)
+                return
+
+    def sectionNameFromSchema(self, schemaTypeName):
+        '''Return a section name from the input schema type name. This section
+        name is a pretty name used in the UI.'''
+
+        # List of special rules for adjusting the base schema names.
+        prefixToAdjust = {
+            'UsdAbc' : '',
+            'UsdGeomGprim' : 'GeometricPrim',
+            'UsdGeom' : '',
+            'UsdHydra' : '',
+            'UsdImagingGL' : '',
+            'UsdLux' : '',
+            'UsdMedia' : '',
+            'UsdRender' : '',
+            'UsdRi' : '',
+            'UsdShade' : '',
+            'UsdSkelAnimation' : 'SkelAnimation',
+            'UsdSkelBlendShape': 'BlendShape',
+            'UsdSkelSkeleton': 'Skeleton',
+            'UsdSkelRoot' : 'SkelRoot', 
+            'UsdUI' : '',
+            'UsdUtils' : '',
+            'UsdVol' : ''
+        }
+        for p, r in prefixToAdjust.items():
+            if schemaTypeName.startswith(p):
+                schemaTypeName = schemaTypeName.replace(p, r, 1)
+                break
+
+        # Put a space in the name when proceeded with a capital letter.
```

Comment:
Not quite sure here, but think you mean either "preceded by" or "followed by".

Reply:
Ah yes, typo